internal/handlers/initrd.go

```diff
@@ -89,6 +93,25 @@ func initrdOverlayReader(imageStore imagestore.ImageStore, client *AssistedServi
 			return nil, "", http.StatusInternalServerError, fmt.Errorf("failed to create append reader for initrd: %v", err)
 		}
 
+		versionOK, err := common.VersionGreaterOrEqual(version, isoeditor.MinimalVersionForNmstatectl)
+		if err != nil {
+			return nil, "", http.StatusInternalServerError, err
+		}
+
+		if versionOK {
+			nmstatectlPath := imageStore.NmstatectlPathForParams(version, arch)
+			if filepath.Dir(nmstatectlPath) != imageStore.GetDataDir() {
```

Comment:
Instead of creating this getter, how about we return an error from `NmstatectlPathForParams` if the path ends up escaping the data directory?